logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -75,9 +64,9 @@ internal class ConnectionDataSource(
     private val connectionApi: ConnectionApi,
     private val userDetailsApi: UserDetailsApi,
     private val userDAO: UserDAO,
-    private val metadataDAO: MetadataDAO,
+    private val selfUser: UserId,
+    private val selfUserTeamId: SelfTeamIdProvider,
```

Comment:
selfUserTeamId, but you are returning SelfTeamIdProvider

Reply:
This provider is a function interface. It is pretty much a lambda.

I guess we could name it `provideSelfUserTeamId` instead.

This way we can call like a function:

```kotlin
provideSelfUserTeamId()

// Instead of

selfUserTeamId()

// Or

selfUserTeamIdProvider.invoke()
```